tests/unit/maintainers_test.sh

```diff
@@ -109,16 +112,157 @@ function test_maintainers_main()
   multilineAssertEquals "$CORRECT_TMP_FS_MSG" "$ret"
 
   cd fs || {
-    fail "($LINENO) It was not possible to move to fs directory"
+    fail "(${LINENO}) It was not possible to move to fs directory"
     return
   }
   ret="$(maintainers_main ..)"
   multilineAssertEquals "$CORRECT_TMP_MSG" "$ret"
 
   ret="$(maintainers_main .)"
-  multilineAssertEquals "$CORRECT_TMP_FS_MSG" "$ret"
+  multilineAssertEquals "$ret" "$CORRECT_TMP_FS_MSG"
+
+  cd "$original_dir" || {
+    fail "(${LINENO}) It was not possible to move back from temp directory"
+    return
+  }
+  rm -rf "$SHUNIT_TMPDIR"
+  mkdir -p "$SHUNIT_TMPDIR"
+}
+
+# This function tests expected behaviors for failure in
+# parse_maintainers_options.
+function test_parse_maintainers_options()
+{
+  local original_dir="$PWD"
+  cd "$FAKE_KERNEL" || {
+    fail "(${LINENO}) It was not possible to move to temporary directory"
+    return
+  }
+
+  # Invalid option (typo)
+  unset options_values
+  declare -gA options_values
+  parse_maintainers_options --update-pacth # Messy argument
+  ret="$?"
+  assertEquals "(${LINENO})" 22 "$ret"
+
+  # Invalid option (even with a correct one following it)
+  unset options_values
+  declare -gA options_values
+  parse_maintainers_options --update-patch --invalid-option
+  ret="$?"
+  assertEquals "(${LINENO})" 22 "$ret"
+
+  # Too many arguments
+  unset options_values
+  declare -gA options_values
+  parse_maintainers_options --update-patch '.' 'some_file'
+  ret="$?"
+  assertEquals "(${LINENO})" 22 "$ret"
+
+  cd "$original_dir" || {
+    fail "(${LINENO}) It was not possible to move back from temp directory"
+    return
+  }
+}
+
+function test_maintainers_main_update_patch_invalid_inputs()
+{
+  local ret
+  local -r original_dir="$PWD"
+
+  cd "$FAKE_KERNEL" || {
+    fail "(${LINENO}) It was not possible to move to temporary directory"
+    return
+  }
+
+  msg="$(maintainers_main --update-patch 'fs/some_file')"
```

Comment:
Declare this variable as local.
Also, use a more intuitive name, something like output_from_function_xpto...

P.S.: Please review this entire commit with this comment in mind.